modules/util.py

```diff
@@ -62,6 +62,22 @@ def getBootPartition(conffile):
         return bootdevice
     return None
 
+def getPartitionRelativeToBoot(conffile, label, relativeIndex):
+    ''' Returns the partition device path when index is relative to boot partition '''
+    # First search by label
+    partdevice = getDevice(label)
+    if not partdevice:
+        match = re.match(r"(.*?)(\d+$)", getBootPartition(conffile))
```

Comment:
How exactly done with pyparted? I explained in the commit log why I didn't use nextPartitions(). I could use the partitions field of the object but given the fixed setup of our partition table I found it easier and simpler to understand.